test/utils/general.spec.js

```diff
@@ -169,30 +169,30 @@ describe('services', () => {
       it('processes an array where the index field is a single value', () => {
         const inputArray = [
           { category: '1', value: 'a' },
-          { category: '1', value: 'b' }
+          { category: '1', value: 'b' },
         ];
-        expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({ '1': inputArray });
+        expect(GeneralUtil.indexByField(inputArray, 'category')).toStrictEqual({ 1: inputArray });
```

Comment:
It should preserve the original type here, the function does not convert the type from string to int.